include/nanoeigenpy/decompositions/eigen-solver.hpp

```diff
@@ -15,54 +15,59 @@ void exposeEigenSolver(nb::module_ m, const char *name) {
   using MatrixType = _MatrixType;
   using Solver = Eigen::EigenSolver<MatrixType>;
 
-  nb::class_<Solver>(m, name, "Eigen solver.")
+  if (!register_symbolic_link_to_registered_type<Solver>(m)) {
```

Comment:
I suggest the following:
```suggestion
  if (register_symbolic_link_to_registered_type<Solver>(m)) {
    return;
  }
  // ... expose Solver here
```
so that you don't add a huge diff just for indentation.